Sources/core/objects/Gun.cpp

```diff
@@ -0,0 +1,96 @@
+#include "Gun.h"
+#include "Tile.h"
+
+Gun::Gun(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+    ammunition_ = 1;
+    max_ammunition_ = 1;
+}
+
+bool Gun::UseAmmo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Gun::Shoot(VDir target)
+{
+    id_ptr_on<CubeTile> tile = GetOwner()->GetOwner();
```

Comment:
Laser has more than one sound so in that case the arguments would require one more.I made the vector as a container for sounds and in every gun it can be set up, like in this case this would be variable like max_ammuntion_ . I agree that it was a mistake to make them shoot argument.
```
class Gun: public Item
{
public:
    virtual void Shoot(VDir target,const std::string& projectile_type,int rand_number);
protected:
    std::vector<std::string> KV_SAVABLE(sounds);
    std::string KV_SVAABLE(projectile_type);
    int KV_SVAABLE(sound_number);
};

void Gun::Shoot(VDir target)
{
    id_ptr_on<CubeTile> tile = GetOwner()->GetOwner();
    if (tile.valid())
    {	
        if (UseAmmo())
        { 
            auto projectile = Create<Projectile>(projectile_type,tile);
            projectile->MakeMovementPattern(target,GetOwner());	
            unsigned int value = GetRand() % sound_number;
            PlaySoundIfVisible(sounds[value], tile.ret_id());
            
        }
        else
        {
            PlaySoundIfVisible(sounds[(sounds.size() - 1)],tile.ret_id()); // last_element will be always the empty.ogg
            // the *click* text from ss13
        }
    }	
}
```